File: 파이썬/3.스택,큐/10773.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/10773
'''

k = int(input())
l = []
for i in range(k) :
    m = int(input())
    if m == 0 :
        l.pop()
    else :
        l.append(m)

print(sum(l))

File: C++/7. 스택 - 초급/10773.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/10773
*/

#include <iostream>
#include <stack>

using namespace std;

int n, t, sum;
stack<int> s;

int main(){

    cin >> n;
    
    for(int i=0; i<n; i++){
        cin >> t;
        if(t == 0){
            s.pop();
        }else{
            s.push(t);
        }
    }
    
    while(!s.empty()){
        sum += s.top();
        s.pop();
    }
    
    cout << sum;
    
    return 0;
}